junit/kube-api-test/core/src/test/java/io/fabric8/kubeapitest/sample/KubernetesMutationHookHandlingTest.java

```diff
@@ -0,0 +1,224 @@
+import io.fabric8.kubeapitest.junit.KubeConfig;
+import io.fabric8.kubernetes.api.model.admission.v1.AdmissionReview;
+import io.fabric8.kubernetes.api.model.admissionregistration.v1.MutatingWebhookConfiguration;
+import io.fabric8.kubernetes.api.model.networking.v1.*;
+import io.fabric8.kubernetes.client.Config;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClientBuilder;
+import io.fabric8.kubernetes.client.utils.Serialization;
+import io.javaoperatorsdk.webhook.admission.AdmissionController;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.bouncycastle.asn1.x509.GeneralName;
+import org.bouncycastle.cert.X509CertificateHolder;
+import org.bouncycastle.cert.jcajce.JcaX509CertificateConverter;
+import org.bouncycastle.openssl.PEMKeyPair;
+import org.bouncycastle.openssl.PEMParser;
+import org.bouncycastle.openssl.jcajce.JcaPEMKeyConverter;
+import org.eclipse.jetty.server.*;
+import org.eclipse.jetty.server.handler.AbstractHandler;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.Base64;
+import java.util.HashMap;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Test demonstrates how to test locally Mutating Webhooks, in real like you implementation such an
+ * endpoint would be simpler, using frameworks like
+ * <a href="https://github.com/java-operator-sdk/kubernetes-webooks-framework">Kubernetes Webhook
+ * Framework</a> with combination of Quarkus or Spring.
+ */
+@EnableKubeAPIServer
+class KubernetesMutationHookHandlingTest {
+
+  @KubeConfig
+  static String kubeConfig;
+
+  private static final Logger log = LoggerFactory.getLogger(KubernetesMutationHookHandlingTest.class);
+
+  public static final String PASSWORD = "secret";
+  public static final String TEST_LABEL_KEY = "test-label";
+  public static final String TEST_LABEL_VALUE = "mutation-test";
+
+  static File certFile = new File("target", "mutation.crt");
+  // server that handles mutation hooks
+  static Server server = new Server();
+
+  // using https://github.com/java-operator-sdk/kubernetes-webooks-framework framework to implement
+  // the response
+  static AdmissionController<Ingress> mutationController = new AdmissionController<>((resource, operation) -> {
```

Comment:
Are we depending on Java Operator SDK's kubernetes-webhooks-framework for using AdmissionController class?  Could this lead to circular dependency issues? Shall we port the required class to Fabric8?

Reply:
The core of Kube API Server does not use the client itself. I don't see how just porting that class would help.